hack/e2e-test.sh

```diff
@@ -42,19 +95,42 @@ trap 'teardown' EXIT
 # TODO: Determine how to wait for readiness of the validation webhook
 sleep 120
 
-echo "Creating ClusterDeployment ${CLUSTER_NAME}"
-
-oc process -f config/templates/cluster-deployment.yaml \
-   CLUSTER_NAME="${CLUSTER_NAME}" \
-   SSH_KEY="${SSH_PUB_KEY}" \
-   PULL_SECRET="${PULL_SECRET}" \
-   AWS_ACCESS_KEY_ID="${AWS_ACCESS_KEY_ID}" \
-   AWS_SECRET_ACCESS_KEY="${AWS_SECRET_ACCESS_KEY}" \
-   BASE_DOMAIN="${BASE_DOMAIN}" \
-   INSTALLER_IMAGE="${INSTALLER_IMAGE}" \
-   OPENSHIFT_RELEASE_IMAGE="" \
-   TRY_INSTALL_ONCE="true" \
-   | oc apply -f -
+
+i=1
+while [ $i -le ${max_tries} ]; do
+  if [ $i -gt 1 ]; then
+    # Don't sleep on first loop
+    echo "sleeping ${sleep_between_tries} seconds"
+    sleep ${sleep_between_tries}
+  fi
+
+  echo "Creating ClusterDeployment ${CLUSTER_NAME}. Try #${i}/${max_tries}:"
+  if oc process -f config/templates/cluster-deployment.yaml \
+         CLUSTER_NAME="${CLUSTER_NAME}" \
+         SSH_KEY="${SSH_PUB_KEY}" \
+         PULL_SECRET="${PULL_SECRET}" \
+         AWS_ACCESS_KEY_ID="${AWS_ACCESS_KEY_ID}" \
+         AWS_SECRET_ACCESS_KEY="${AWS_SECRET_ACCESS_KEY}" \
+         BASE_DOMAIN="${BASE_DOMAIN}" \
+         INSTALLER_IMAGE="${INSTALLER_IMAGE}" \
+         OPENSHIFT_RELEASE_IMAGE="" \
+         TRY_INSTALL_ONCE="true" \
+         | oc apply -f - ; then
```

Comment:
What can cause this to fail intermittently?

Reply:
If the API server is down when the `oc apply` is run. For instance, if the certs are updated and the api server pods are rolled (they're talking about having really short cert periods for CI to test the rolling).

Or if above where `CLUSTER_NAME` is gathered, if that `oc` call fails (although now it'll retry and ultimately error out). 

This is the actual line that my [e2e was failing](https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_hive/219/pull-ci-openshift-hive-master-e2e/59) on and then @csrwng pointed out that we should add retries to the other `oc` commands as well.

Reply:
Sounds good thanks.